src/nvim/change.c

```diff
@@ -129,6 +129,7 @@ void changed(void)
 void changed_internal(void)
 {
   curbuf->b_changed = true;
+  curbuf->b_changed_notified = false;
```

Comment:
Usually we prefer invalidiation flags to be the other way around, i e a flag that is `true` if a later action is needed.